beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -213,17 +238,19 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
     }
 
     /// A request to find peers on a given subnet.
-    pub fn discover_subnet_peers(&mut self, subnet_id: SubnetId, min_ttl: Option<Instant>) {
+    pub fn discover_subnet_peers(&mut self, subnets_to_discover: Vec<SubnetDiscovery>) {
```

Comment:
I may be missing something here. 
In the previous version we built the custom `eq()` implementation which is now on `SubnetDiscovery`. It was used to ensure we were not requesting duplicate subnet queries. 
If we have a few queued, I can't seem to see if there is a check that prevents us from adding duplicate subnet queries?

Reply:
Ahh you are right. I had it as `BTreeSet<SubnetDiscovery>` from the attestation service but changed it later on. 
Will fix this to prevent duplicate queries. Thanks!

Reply:
Did a little more digging. In the earlier implementation, we were preventing duplicate queries from being made here
https://github.com/sigp/lighthouse/blob/master/beacon_node/network/src/attestation_service/mod.rs#L366
the custom eq implementation is useful in the attestation service tests.

Made changes to prevent duplicate queries from reaching discovery in this commit https://github.com/sigp/lighthouse/pull/1415/commits/e8393a284188d6cdea7a85ced842eada99a40e28

Reply:
Yep this is good. It would also be nice to prevent duplicates from being able to queue inside the discovery service also. In case we get different subscription API calls that create duplicate discoveries. 

If we ensure the discovery queue is clean also when adding them we can be sure we're not doing duplicate discovery queries. 

This is somewhat important because discovery queries are very time consuming

Reply:
Isn't the duplicate handling in discovery done here? https://github.com/sigp/lighthouse/blob/e8393a284188d6cdea7a85ced842eada99a40e28/beacon_node/eth2_libp2p/src/discovery/mod.rs#L396-L409

If there's a duplicate in the queue, we simply update the min_ttl and retries for that subnet query.

Reply:
Yep, forgot that guy was there. Thanks